pgrx/src/pg_sys.rs

```diff
@@ -14,11 +15,13 @@ pub use pgrx_pg_sys::*;
 */
 #[inline]
 pub unsafe fn rt_fetch(index: Index, range_table: *mut List) -> *mut RangeTblEntry {
-    crate::list::List::<*mut core::ffi::c_void>::downcast_ptr(range_table)
-        .expect("rt_fetch used on non-ptr List")
-        .get((index - 1) as _)
-        .expect("rt_fetch used out-of-bounds")
-        .cast()
+    mem::current_context(|cx| {
+        crate::list::List::<*mut core::ffi::c_void>::downcast_ptr_in_memcx(range_table, cx)
+            .expect("rt_fetch used on non-ptr List")
+            .get((index - 1) as _)
+            .expect("rt_fetch used out-of-bounds")
+            .cast()
+    })
```

Comment:
This diff suggests to me my approach is kinda incorrect, since I don't really "use" the context.